api-spec.yaml

```diff
@@ -0,0 +1,482 @@
+      security:
+        - library_auth:
+            - write:books
+            - read:books
+    post:
+      tags:
+        - books
+      summary: Add a new book to the library
+      operationId: addBook
+      requestBody:
+        description: Create a new book in the library
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Book'
+        required: true
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Book'
+
+        '405':
+          description: Invalid input
+      security:
+        - library_auth:
+            - write:books
+            - read:books
+    get:
+      summary: Search for books by title and/or author
+      description: Returns a list of books matching the search criteria
+      parameters:
+        - in: query
+          name: title
+          schema:
+            type: string
+          description: Title of the book to search for
+        - in: query
+          name: author
+          schema:
+            type: string
+          description: Author of the book to search for
+      responses:
+        '200':
+          description: A list of books matching the search criteria
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Book'
+        '400':
+          description: Invalid ID supplied
+      security:
+        - api_key: [ ]
+        - library_auth:
+            - write:books
+            - read:books
+  /books/{bookId}:
+    get:
+      tags:
+        - books
+      summary: Find book by ID
+      description: Returns a single book
+      operationId: getBookById
+      parameters:
+        - name: bookId
+          in: path
+          description: ID of book to return
+          required: true
+          schema:
+            type: integer
+            format: int64
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Book'
+
+        '400':
+          description: Invalid ID supplied
+        '404':
+          description: Book not found
+      security:
+        - api_key: []
+        - library_auth:
+            - write:books
+            - read:books
+    delete:
+      tags:
+        - books
+      summary: Deletes a book
+      description: delete a book
+      operationId: deleteBook
+      parameters:
+        - name: bookId
+          in: path
+          description: ID of book to return
+          required: true
+          schema:
+            type: integer
+            format: int64
+      responses:
+        '400':
+          description: Invalid book value
+      security:
+        - library_auth:
+            - write:books
+            - read:books
+  /books/{bookId}/cover:
```

Comment:
Since this endpoint returns `Url` we can add the url as a field to the `Book` schema. 
or we can return the image (content-type: image/jpeg) here.